### PR TITLE
First pass at custom handler extension

### LINKED ISSUE #1: Pass URL into _onSubmit()
@jywarren made this suggestion on the plots-dev mailing list:

> I wonder if _fireCreatedEvent and _onSubmit could be made more modular so that it could be tied to a passed URL parameter rather than having to be pasted in. The default could still be a HTML form. Maybe add a 'url' param to the_onSubmit method and make it clear that _fireCreatedEvent can be overridden, or even accept a replacement in the constructor?

I agree with this strategy :+1:

### PR DESCRIPTION
**Pull request by nateemerson**
Fixes #1

### CODE REVIEW

**Comment by nateemerson** [comment]
@jywarren Would be interested in your feedback on this update, as it came from your recommendation (and I'm not super confident in my javascript/leaflet skills)

File: src/leaflet.drawDistortableImage.js [hunk 1]
```diff
@@ -100,27 +100,35 @@ L.Draw.DistortableImage = L.Draw.Rectangle.extend({
...
+                that._onSubmit(document.getElementById('imgUrlInput').value);
+            };
+        }
 
         this._map
             .on('click', this._cancelCreate, this);
     },
 
-    _onSubmit: function() {
+    _onSubmit: function(url) {
```
Comment (ebollens) [inline src/leaflet.drawDistortableImage.js:131]: If this is being contributed back to the original plugin, this sort of interface might preclude it from some version releases. I recommend adding an explicit case within the method for when `typeof url === 'undefined'`, such as:

``` js
if(typeof url === 'undefined')
  url = $('#imgUrlInput').val()
```

Comment (nateemerson) [inline src/leaflet.drawDistortableImage.js:131]: The `_onSubmit` function was introduced by me in this plugin, so it's not referenced/used anywhere else. The default case is provided in the code above (lines 119-125) where it uses the value of `imgUrlInput` in the call to `_onSubmit`.

**Comment by jywarren** [comment]
Checking this out now!

File: src/leaflet.drawDistortableImage.js (hunk 1 above)
Comment (jywarren) [inline src/leaflet.drawDistortableImage.js:131]: FWIW, I like moving any explicit reference to a DOM element into the same area of the code as that element was created, so if that default DOM interface is modified, all refs to it are contained in that section of the code. It's one step closer to a separation of view and logic.

**Comment by jywarren** [comment]
+1 -- readable and improves view/logic abstraction.

**Pull request merged**